### DISCUSSION TITLE
Update Composer packages with wildcard using regex manager

### DISCUSSION THREAD

**Discussion by mbrodala** [renovatebot/renovate#20397] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Renovate proposes an update for all `typo3/*` packages in a single update PR.

### What you tried so far.

Given the following `renovate.json`:

<details>

<summary>renovate.json</summary>

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    ":disableDependencyDashboard",
    ":separatePatchReleases"
  ],
  "commitMessagePrefix": "[TASK] ",
  "commitMessageTopic": "{{depName}}",
  "commitMessageExtra": " ({{{displayFrom}}} => {{{displayTo}}})",
  "composerIgnorePlatformReqs": null,
  "platformAutomerge": true,
  "rangeStrategy": "update-lockfile",
  "packageRules": [
    {
      "matchDatasources": ["docker"],
      "matchPackageNames": ["thecodingmachine/php"],
      "enabled": false
    },
    {
      "matchPackageNames": ["php"],
      "enabled": false
    },
    {
      "matchPackagePrefixes": [
        "typo3/"
      ],
      "matchManagers": ["composer"],
      "enabled": false
    },
    {
      "matchPackagePrefixes": [
        "typo3/"
      ],
      "matchManagers": ["regex"],
      "commitMessageTopic": "TYPO3 CMS",
      "labels": [
        "typo3"
      ],
      "extends": [
        ":disableMajorUpdates"
      ]
    }
  ],
  "regexManagers": [
    {
      "fileMatch": ["^composer.lock$"],
      "matchStrings": [
        "\"name\": \"typo3\\/cms-core\",\\n\\s+\"version\": \"(?<currentValue>[^\"]+)\""
      ],
      "depNameTemplate": "typo3/*",
      "packageNameTemplate": "typo3/cms-core",
      "datasourceTemplate": "packagist"
    }
  ]
}
```

</details>

Renovate correctly proposes a [single update PR](https://github.com/mbrodala/renovate-typo3-regex/pull/2) `[TASK] Update TYPO3 CMS (v11.5.22 => v11.5.x)`:

![image](https://user-images.githubusercontent.com/5037116/218700882-39aebf54-f075-4faf-90d9-323e138fa092.png)

However, the [actual update](https://github.com/mbrodala/renovate-typo3-regex/pull/2/files) literally only covers `typo3/cms-core` instead of all `typo3/*` packages:

![image](https://user-images.githubusercontent.com/5037116/218701075-8360765c-46e7-4721-a9f0-60be68f38281.png)

Is there a way to adjust the regex manager to have `composer update` invoked with `typo3/*` as desired? (See [official docs](https://docs.typo3.org/m/typo3/guide-installation/11.5/en-us/Minor/Index.html#execute-the-upgrade))

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by RahulGautamSingh** [comment]
It is because you have matched only against `cms-core` in your regex, instead you need to use something that matches all deps with prefix `typo3/*`
Eg.    ``` "\"name\": \"typo3\\/.*\",\\n\\s+\"version\": \"(?<currentValue>[^\"]+)\"" ```

Reproduced it here...
https://github.com/RahulGautamSingh-testing/renovate-typo3-regex/pull/1/files

**Reply by RahulGautamSingh** [reply]
Btw isn't upating the lockfile kind-of manually instead of using the packageManager risky?

**Reply by mbrodala** [reply]
I don't intend to do anything manually. I just want Renovate to pick up the update of `typo3/cms-core` and run `composer update typo3/* --with-all-dependencies`.

**Reply by RahulGautamSingh** [reply]
Ohh! I misunderstood the goal, my bad.

**Reply by mbrodala** [reply]
But evidently you did understand what `regex` does and what not compared to me. :sweat_smile:

**Comment by rarkins** [comment]
No way to invoke a package manager from a regex manager update

**Reply by mbrodala** [reply]
Hm, so my basic assumption is flawed? That would explain why I failed to see an `composer update` command in the Renovate logs.

So this won't work at all? This would have been one solution to https://github.com/renovatebot/renovate/issues/20288
